tests/integration/integration_test.go

```diff
@@ -1146,10 +1146,58 @@ func TestStackReferencePython(t *testing.T) {
 		Config: map[string]string{
 			"org": os.Getenv("PULUMI_TEST_OWNER"),
 		},
+		EditDirs: []integration.EditDir{
+			{
+				Dir:      "step1",
+				Additive: true,
+			},
+			{
+				Dir:      "step2",
+				Additive: true,
+			},
+		},
 	}
 	integration.ProgramTest(t, opts)
 }
 
+func TestMultiStackReferencePython(t *testing.T) {
+	if owner := os.Getenv("PULUMI_TEST_OWNER"); owner == "" {
+		t.Skipf("Skipping: PULUMI_TEST_OWNER is not set")
+	}
+
+	// build a stack with an export
+	exporterOpts := &integration.ProgramTestOptions{
+		Dir: filepath.Join("stack_reference_multi", "python", "exporter"),
+		Dependencies: []string{
+			filepath.Join("..", "..", "sdk", "python", "env", "src"),
+		},
+		Quick: true,
+		Config: map[string]string{
+			"org": os.Getenv("PULUMI_TEST_OWNER"),
+		},
+		NoParallel:       true,
+		SkipStackRemoval: true, // do this at the end after the importer finishes
+	}
+	pt := integration.ProgramTest(t, exporterOpts)
+
+	exporterStackName := exporterOpts.GetStackName().String()
+
+	importerOpts := &integration.ProgramTestOptions{
+		Dir: filepath.Join("stack_reference_multi", "python", "importer"),
+		Dependencies: []string{
+			filepath.Join("..", "..", "sdk", "python", "env", "src"),
+		},
+		Quick: true,
+		Config: map[string]string{
+			"org":                 os.Getenv("PULUMI_TEST_OWNER"),
+			"exporter_stack_name": exporterStackName,
+		},
+		NoParallel: true,
+	}
+	integration.ProgramTest(t, importerOpts)
+	pt.TestLifeCycleDestroy()
```

Comment:
The idiomatic was to do cleanup like this is to wrap in a defer like how we do in `TestLifeCycleInitAndDestroy`

https://github.com/pulumi/pulumi/blob/master/pkg/testing/integration/program.go#L884-L889

That way the cleanup [still runs if you hit a panic](https://blog.golang.org/defer-panic-and-recover) or unexpected error along the way.